packages/ui/src/views/pipeline-edit/components/graph-implementation/types/problem-type.ts

```diff
@@ -0,0 +1,11 @@
+export type ProblemSeverity = 'error' | 'warning' | 'info'
+
+export interface Problem<T = unknown> {
+  severity: ProblemSeverity
+  message: string
+  position: {
```

Comment:
position should be optional, so that we can add non-yaml problems to the Problems component later as well